src/js/components/Calendar/Calendar.js

```diff
@@ -276,75 +240,12 @@ const Calendar = forwardRef(
       if (activeDateProp) setActiveDate(activeDateProp);
     }, [activeDateProp]);
 
-    let timestamp = useMemo(() => timestampProp, [timestampProp]);
-    if (timestampProp === undefined) {
-      if (Array.isArray(dateProp) && dateProp.length) {
-        if (Array.isArray(dateProp[0])) {
-          timestamp = getTimestamp(dateProp[0][0]);
-        } else timestamp = getTimestamp(dateProp[0]);
-        // check to see if value is not an empty string
-        // empty string should behave like undefined
-      } else if (typeof dateProp === 'string' && dateProp.length) {
-        timestamp = getTimestamp(dateProp);
-      } else if (Array.isArray(datesProp) && datesProp.length) {
-        if (Array.isArray(datesProp[0])) {
-          timestamp = getTimestamp(datesProp[0][0]);
-        } else timestamp = getTimestamp(datesProp[0]);
-      } else if (typeof datesProp === 'string') {
-        timestamp = getTimestamp(datesProp);
-      } else if (typeof referenceProp === 'string') {
-        timestamp = getTimestamp(referenceProp);
-      }
-    }
-
-    const normalizedDate = useMemo(
-      () =>
-        timestampProp === undefined
-          ? normalizeDate(dateProp, timestamp, normalize)
-          : dateProp,
-      [dateProp, normalize, timestamp, timestampProp],
-    );
-
-    const normalizedDates = useMemo(
-      () =>
-        timestampProp === undefined
-          ? normalizeDate(datesProp, timestamp, normalize)
-          : datesProp,
-      [datesProp, normalize, timestamp, timestampProp],
-    );
-
-    // set date when caller changes it, allows us to change it internally too
-    const [date, setDate] = useState(normalizedDate);
-    useEffect(() => {
-      setDate(normalizedDate);
-    }, [normalizedDate]);
+    const [date, setDate] = useState(dateProp);
 
-    // set dates when caller changes it, allows us to change it internally too
-    const [dates, setDates] = useState(normalizedDates);
-    useEffect(() => {
-      setDates(normalizedDates);
-    }, [normalizedDates]);
+    const [dates, setDates] = useState(datesProp);
```

Comment:
If the dateProp changes on a future render this won't get updated and it will continue to use the original value unless something else calls setDates. Is that intended?

Reply:
Good callout -- we'll need to look into this. Still exploring behavior here in general, thanks for the note.